astropy/coordinates/baseframe.py

```diff
@@ -312,8 +318,8 @@ def __init__(self, *args, copy=True, representation=None,
                                      'velocity differential is supported. Got: '
                                      '{0}'.format(diffs))
 
-        elif self.representation:
-            representation_cls = self.representation
+        elif self.representation_type:
+            representation_type = self.representation_type
```

Comment:
I think here it is actually clearer to stick with `representation_cls = self.get_representation_cls()` (i.e., use the getter), because the whole stanza below deals with something known to be a class.